src/Parser.js

```diff
@@ -371,10 +403,54 @@ Parser.prototype.parseImplicitGroup = function(pos, mode) {
         } else {
             throw new ParseError("Missing \\right", this.lexer, body.position);
         }
+    } else if (func === "\\begin") {
+        // begin...end is similar to left...right
+        var begin = this.parseFunction(pos, mode);
+        var envName = begin.result.value.name;
+        if (!environments.hasOwnProperty(envName)) {
+            throw new ParseError(
+                "No such environment: " + envName,
+                this.lexer, begin.result.value.namepos);
+        }
+        // Build the environment object. Arguments and other information will
+        // be made available to the begin and end methods using properties.
+        var env = new environments[envName](envName, this);
+        env.parser = this;
+        env.lexer = this.lexer;
+        env.mode = mode;
+        env.beginNode = begin;
+        env.args = [];
```

Comment:
If all of these things are needed by an environment, why not just pass them as arguments to the constructor.  That way people won't forget to set these values.